typing/typemod.ml

```diff
@@ -1398,6 +1510,22 @@ and transl_signature env sg =
             mksig (Tsig_modtype mtd) env loc :: trem,
             sg :: rem,
             final_env
+        | Psig_modtypesubst pmtd ->
+            let info id tmty =
+              let mty = match tmty with
+                | Some tmty -> tmty.mty_type
+                | None -> assert false (* unparsable *)
```

Comment:
This could still come from a ppx and I think we usually try to do better than an `assert false` in such cases.

Reply:
True. Since this seemed to be a parsing error, I made this requirement an ast invariant. I thought about using using a hand-tailored type for the right hand side of module type declaration, but that required quite a few changes to the various parsetree mappers and iterators with didn't seem really useful. .